File: CodeEval/easy/019bitpositions/solution.py
#!/usr/bin/env python

from sys import argv

bool = {True: 'true', False: 'false'}

for line in open(argv[1], 'r'):
    n, i, j = map(int, line.strip().split(","))
    print(bool[(n >> i - 1) & 1 == (n >> j - 1) & 1])

File: codeeval/easy/019bitpositions/solution.rb
#!/usr/bin/env ruby

File.foreach(ARGV[0]).each do |line|
  n,i,j = line.strip.split(",").map { |x| x.to_i }
  a = (n >> i-1) & 1
  b = (n >> j-1) & 1
  puts a == b ? "true" : "false"
end
